jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/vsdm/VsdmReceiver.java

```diff
@@ -35,75 +37,130 @@
 
-		while (true) {
-			
-			DatagramPacket packet = new DatagramPacket(new byte[odeProperties.getVsdmBufferSize()], odeProperties.getVsdmBufferSize());
+		logger.info("Vsdm Receiver Service started.");
+
+		byte[] buffer = new byte[odeProperties.getVsdmBufferSize()];
+
+		DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+
+		Boolean stopped = false;
+		while (!stopped) {
 
 			try {
+				logger.info("VSDM RECEIVER - Waiting for UDP packets...");
 				socket.receive(packet);
+				logger.info("VSDM RECEIVER - Packet received.");
+				String obuIp = packet.getAddress().getHostAddress();
+				int obuPort = packet.getPort();
+				logger.info("Packet length: {}, Buffer length: {}", packet.getLength(), buffer.length);
+				byte[] actualPacket = Arrays.copyOf(packet.getData(), packet.getLength());
+				if (packet.getLength() > 0) {
+					logger.info("VSDM RECEIVER - Received data:", buffer);
+					decodeData(actualPacket, obuIp, obuPort);
+				}
 			} catch (IOException e) {
-				logger.error("[VSDM Receiver] Error receiving UDP packet", e);
-			}
-
-			InputStream ins = new ByteArrayInputStream(packet.getData());
-			AbstractData decoded = null;
-			try {
-				decoded = coder.decode(ins, decoded);
-			} catch (DecodeFailedException | DecodeNotSupportedException e) {
-				logger.error("[VSDM Receiver] Error, unable to decode UDP message", e);
+				logger.error("VSDM RECEIVER - Error receiving UDP packet", e);
+				stopped = true;
 			}
+		}
+	}
 
-			if (decoded instanceof ServiceRequest || decoded instanceof ServiceResponse) {
-				// send
+	private void decodeData(byte[] msg, String obuIp, int obuPort) {
+		try {
+			AbstractData decoded = J2735Util.decode(coder, msg);
+			logger.info("VSDM RECEIVER - Decoded the message");
+			if (decoded instanceof ServiceRequest) {
+				logger.info("VSDM RECEIVER - Received ServiceRequest: ", decoded.toString());
+				ServiceRequest request = (ServiceRequest) decoded;
+				ReqResForwarder forwarder = new ReqResForwarder(odeProperties, request, obuIp, obuPort);
+				Thread forwarderThread = new Thread(forwarder, "forwarderThread");
+				forwarderThread.start();
 			} else if (decoded instanceof VehSitDataMessage) {
-				// send 
-				List<BasicSafetyMessage> bsmList = VsdToBsmConverter.convert((VehSitDataMessage) decoded);
-				for (BasicSafetyMessage entry : bsmList) {
-					try {
-						J2735Bsm convertedBsm = OssBsm.genericBsm(entry);
-						String bsmJson = JsonUtils.toJson(convertedBsm, odeProperties.getVsdmVerboseJson());
-				        
-						publish(bsmJson);
-
-		        logger.debug("Published: {}", bsmJson);
-					} catch (OssBsmPart2Exception e) {
-						logger.error("[VSDM Receiver] Error, unable to convert BSM: ", e);
-					}
-
-				}
-
+				logger.info("VSDM RECEIVER - Received VSD");
+				
+				logger.info("VSDM RECEIVER - Forwarding VSD to SDC");
+				VsdDepositor depositor = new VsdDepositor(odeProperties, msg);
+				Thread depositorThread = new Thread(depositor, "depositor");
```

Comment:
VSDM transfer from receiver to depositor will probably be done via Kafka next refactor, but we probably still want a thread to start for each new ServiceRequest. Or maybe we could use Kafka to route those to some sort of session manager service.